File: 0690-employee-importance/0690-employee-importance.cpp
/*
// Definition for Employee.
class Employee {
public:
    int id;
    int importance;
    vector<int> subordinates;
};
*/

class Solution {
public:
    int getImportance(vector<Employee*> employees, int id) {
        vector<vector<int>>adj(2000);
        vector<int>imp(2000);
        for(auto &it:employees){
            imp[it->id-1]=it->importance;
            for(int sub:it->subordinates){
                adj[it->id-1].push_back(sub-1);
            }
        }
        queue<int>q;
        q.push(id-1);
        int ans=imp[id-1];
        while(!q.empty()){
            int node=q.front();
            q.pop();
            for(int nb:adj[node]){
                ans+=imp[nb];
                q.push(nb);
            }
        }
        return ans;
    }
};

File: 0690-employee-importance/0690-employee-importance.py
"""
# Definition for Employee.
class Employee(object):
    def __init__(self, id, importance, subordinates):
    	#################
        :type id: int
        :type importance: int
        :type subordinates: List[int]
        #################
        self.id = id
        self.importance = importance
        self.subordinates = subordinates
"""

from collections import deque

class Solution(object):
    def getImportance(self, employees, id):

        adj = {}
        imp = {}

        # Build graph
        for emp in employees:
            imp[emp.id] = emp.importance
            adj[emp.id] = emp.subordinates


        q = deque()
        q.append(id)

        ans = 0

        while q:

            node = q.popleft()

            ans += imp[node]

            for sub in adj[node]:
                q.append(sub)

        return ans

        